File: Java/basics/DataTypes.java
package basics;

public class DataTypes {
    public static void main(String[] args) {
        // -128 to 127 --> 8 bits
        byte numByte = 100;
        System.out.println(numByte);

        // -32,708 to 32,767 --> 16 bits
        short numShort = 32000;
        System.out.println(numShort);

        // -2^31 to 2^31 - 1 --> 32 bits
        int numInt = 100000000;
        System.out.println(numInt);
        
        // -2^63 to 2^63 - 1 --> 64 bits
        long numLong = 1000000000;
        System.out.println(numLong);

        // upto 6-7 decimal places --> 32 bits
        float numFloat = 5.123456f;
        System.out.println(numFloat);

        // upto 15 decimal places --> 64 bits
        double numDouble = 5.123456789123456;
        System.out.println(numDouble);

        // All 256 characters
        char ch = 'c';
        System.out.println(ch);

        // true or false
        boolean fl = true;
        System.out.println(fl);
    }
}

File: Basics/data_types.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    // short --> 2 bytes
    short short_num = 32767;
    cout << short_num << endl;

    // int --> 2 or 4 bytes
    int int_num = 2147483647;
    cout << int_num << endl;

    // float --> upto 5-7 decimal places
    float float_num = 1.12345;
    cout << float_num << endl;

    // double --> upto 15 decimal places
    double double_num = 1.123456789123456;
    cout << double_num << endl;

    // char --> all 256 characters
    char ch = 'c';
    cout << ch << endl;

    // bool --> true or false
    bool bool_var = true;
    cout << bool_var << endl;

    // long --> 4 or 8 bytes
    long long_num = 2147483647123;
    cout << long_num << endl;

    // long long --> 8 bytes
    long long long_long_num = 9223372036854775807;
    cout << long_long_num << endl;

    return 0;
}